cmd/server/workflow/workflow.proto

```diff
@@ -0,0 +1,147 @@
+syntax = "proto3";
+option go_package = "github.com/argoproj/argo/cmd/server/workflow";
+
+
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "k8s.io/apimachinery/pkg/apis/meta/v1/generated.proto";
+import "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1/generated.proto";
+import "k8s.io/api/core/v1/generated.proto";
+
+// Workflow Service
+//
+// Workflow Service API performs CRUD actions against application resources
+package workflow;
+
+message SubmitOptions {
+    string name = 1;
+    string generateName = 2;
+    string instanceID = 3;
+    string entrypoint = 4;
+    repeated string parameters = 5;
+    string serviceAccount = 6;
+    bool serverDryRun = 7;
+    string labels = 8;
+    k8s.io.apimachinery.pkg.apis.meta.v1.OwnerReference ownerReference = 9;
+}
+
+message WorkflowCreateRequest {
+    string namespace = 1;
+    github.com.argoproj.argo.pkg.apis.workflow.v1alpha1.Workflow workflow = 2;
+    k8s.io.apimachinery.pkg.apis.meta.v1.CreateOptions createOptions = 3;
+    SubmitOptions submitOptions = 4;
+}
+
+message WorkflowGetRequest {
+    string workflowName = 1;
+    string namespace = 2;
+    k8s.io.apimachinery.pkg.apis.meta.v1.GetOptions getOptions = 3;
+}
+
+message WorkflowListRequest {
+    string namespace = 1;
+    k8s.io.apimachinery.pkg.apis.meta.v1.ListOptions listOptions = 2;
+}
+
+message WorkflowUpdateRequest {
+    string workflowName = 1;
+    string namespace = 2;
+    bool memoized = 3;
+}
+
+message WorkflowLogRequest {
+    string workflowName = 1;
+    string namespace = 2;
+    string podName = 3;
+    string container = 4;
+    k8s.io.api.core.v1.PodLogOptions logOptions = 5;
+}
+
+
+message WorkflowDeleteRequest {
+    string workflowName = 1;
+    string namespace = 2;
+    k8s.io.apimachinery.pkg.apis.meta.v1.DeleteOptions deleteOptions = 3;
+}
+message WorkflowDeleteResponse {
+    string workflowName = 1;
+    string status = 2;
+}
+
+message LogEntry {
+    string content = 1;
+    k8s.io.apimachinery.pkg.apis.meta.v1.Time timeStamp = 2;
+}
+
+service WorkflowService {
+    rpc Create (WorkflowCreateRequest) returns (github.com.argoproj.argo.pkg.apis.workflow.v1alpha1.Workflow) {
```

Comment:
In Argo CD, we ended up with unpredictable Swagger operation names when we combined photos. Can I suggest all operation are verb+noun, e.g. "CreateWorkflow" rather than just "Create".